lib/Dialect/TritonGPU/Transforms/OptimizeDotOperands.cpp

```diff
@@ -227,20 +235,32 @@ class UseShmemForScales
     return op.getDefiningOp<Op>();
   }
 
-  bool isDescendingOrder(triton::gpu::MemDescType scale) const {
-    auto order = triton::gpu::getOrder(scale);
-    auto rank = scale.getRank();
-    for (int i = 0; i < rank; ++i) {
-      if (order[i] != rank - 1 - i)
-        return false;
-    }
-    return true;
+  bool isInnermostContiguous(MemDescType type, unsigned numElems) const {
+    ArrayRef<int64_t> shape = type.getShape();
+    Attribute enc = type.getEncoding();
+    MLIRContext *ctx = enc.getContext();
+
+    LinearLayout actual = toLinearLayout(shape, enc);
+    StringAttr fastestIn = *actual.getInDimNames().begin();
+
+    // Flatten actual outs in reverse order to produce a row-major flattening
+    // of the layout
+    auto outNames = actual.getOutDimNames();
+    SmallVector<StringAttr> revOut(outNames.begin(), outNames.end());
+    std::reverse(revOut.begin(), revOut.end());
+    actual = actual.transposeOuts(revOut).flattenOuts();
+
+    LinearLayout tile = LinearLayout::identity1D(
+        numElems, fastestIn, *actual.getOutDimNames().begin());
+
+    return static_cast<bool>(divideLeft(actual, tile));
```

Comment:
Here you can just do
```suggestion
    return actual.getNumConsecutiveInOut() >= numElems;
```